### ISSUE TITLE
Translation appears to have built successfully, but is not visible on website

### ISSUE THREAD

**Issue by jbphet** [phetsims/perennial#85]
I'm keeping an eye on translation submissions, since there have been some issues lately.  I noticed that a translation of plinko-probability was submitted recently for the sv (Swedish) locale, and it doesn't appear to be visible on the website.  Below are some details that should help in tracking down the issue.

This may be related to a previously resolved issue that had somwhat similar symptoms, see https://github.com/phetsims/perennial/issues/72.

Here is an excerpt from rosetta's log where the build request was made:

```
Feb 26 16:54:49 phet-server.int.colorado.edu rosetta[64529]: Feb 26 2018 16:54:49 MST - info: sending build request to server, URL = https://phet.colorado.edu/deploy-html-simulation
Feb 26 16:54:49 phet-server.int.colorado.edu rosetta[64529]: Feb 26 2018 16:54:49 MST - info: build request returned status 202
Feb 26 16:54:49 phet-server.int.colorado.edu rosetta[64529]: Feb 26 2018 16:54:49 MST - info: build request successfully submitted to build server
Feb 26 16:54:49 phet-server.int.colorado.edu rosetta[64529]: Feb 26 2018 16:54:49 MST - info: finished string submission for plinko-probability_sv
```

...and here is the start of the corresponding build in the build-server log:
```
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: info: deploy request received, original URL = http://phet.colorado.edu/deploy-html-simulation
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: api:"2.0"
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: dependencies:<omitted for brevity>
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: simName:"plinko-probability"
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: version:"1.1.2"
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: locales:["sv"]
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: servers:["production"]
Feb 26 16:54:49 phet-server.int.colorado.edu build-server[22220]: brands:["phet"]
```

...and here is the end of the build process in the build-server log:

```
Feb 26 16:55:27 phet-server.int.colorado.edu build-server[22220]: info: build for plinko-probability finished successfully
```

...and, finally, here is a screenshot from https://phet.colorado.edu/en/simulation/plinko-probability (note that no Swedish translation is available):
![image](https://user-images.githubusercontent.com/3082831/36810198-42035f50-1c87-11e8-9dba-4ca9bda5020b.png)

**Comment by jbphet** [comment]
Assigning to @mattpen for initial investigation.

**Comment by mattpen** [comment]
It looks like there may still be some concurrency problems with the request to notify the server about translations.

Here are the logs from the syncronize project request:

```
26 Feb 2018 16:55:26,494  INFO                      sync - Synchronizing project html/plinko-probability with docroot /data/web/static
26 Feb 2018 16:55:26,507 DEBUG                      sync - detecting project with root/data/web/static/phetsims/sims/html/plinko-proba
26 Feb 2018 16:55:26,508 DEBUG                      sync - Project already exists
26 Feb 2018 16:55:26,535 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - en - Plinko Prob
26 Feb 2018 16:55:26,552  WARN                      sync - Found simulation with project html/plinko-probability
26 Feb 2018 16:55:26,552 DEBUG                      sync - Found simulation plinko-probability
26 Feb 2018 16:55:26,558  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,582 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - bs - Plinko vjer
26 Feb 2018 16:55:26,583  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,589 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - da - Plinko-sand
26 Feb 2018 16:55:26,590  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,596 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - de - Plinko Nage
26 Feb 2018 16:55:26,598  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,604 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - el - Πιθανότητες
26 Feb 2018 16:55:26,604  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,611 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - es - Probabilida
26 Feb 2018 16:55:26,612  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,618 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - es_MX - Probabil
26 Feb 2018 16:55:26,619  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,624 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - es_PE - Probabil
26 Feb 2018 16:55:26,625  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,631 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - eu - Plinko Prob
26 Feb 2018 16:55:26,632  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,638 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - fr - Probabilité
26 Feb 2018 16:55:26,639  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,645 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - hu - Galton-desz
26 Feb 2018 16:55:26,646  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,652 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - it - Probabilità
26 Feb 2018 16:55:26,653  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,659 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - kk - Ықтималдықт
26 Feb 2018 16:55:26,662  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,670 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - ko - 플린코 확률
26 Feb 2018 16:55:26,674  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,681 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - lv - Plinko varb
26 Feb 2018 16:55:26,685  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,691 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - mr - प्लिन्को संभाव्
26 Feb 2018 16:55:26,693  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,699 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - nb - Plinko sann
26 Feb 2018 16:55:26,701  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,707 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - nl - Plinko Prob
26 Feb 2018 16:55:26,708  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,713 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - pl - Deska Galto
26 Feb 2018 16:55:26,714  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,720 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - pt_BR - Probabil
26 Feb 2018 16:55:26,721  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,726 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - ru - Элементы те
26 Feb 2018 16:55:26,727  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,733 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - sk - Pravdepodob
26 Feb 2018 16:55:26,734  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,740 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - sr - Плинко веро
26 Feb 2018 16:55:26,741  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,748 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - vi - Xác suất Pl
26 Feb 2018 16:55:26,749  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,755 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - zh_CN - 二项分布
26 Feb 2018 16:55:26,756  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:26,762 DEBUG                      sync - Reading localized simulation XML for: plinko-probability - zh_TW - Plinko P
26 Feb 2018 16:55:26,763  WARN                      sync - Setting iPad and chromebook categories for HTML5 sim
26 Feb 2018 16:55:27,128  INFO         AddHtmlTranslator - Not found:  locale:sv
```

There was apparently no swedish entry in the translations .xml file when the sync call was made, so when the translator was added the call failed.  

This is essentially the same problem that caused https://github.com/phetsims/perennial/issues/72.  I'm investigating some other problems with the build-server at the moment and will take a look at this too.

**Comment by mattpen** [comment]
It looks like sv is still missing from plinko-probability.xml.  The most likely reason this could happen is if `/babel/plinko-probability/plinko-probability-strings_sv.json` did not exist when the build server tried to create the xml file.  The github file was committed on 2/26@16:54 and the xml was built on 2/26@16:55, so the file should have been available to the build-server. I was able to manually add the title entry for swedish to the xml file and deploy it to the website. 

I'm confused how this could have happened.  @jbphet can you check rosetta and see if there is anything obvious?

**Comment by mattpen** [comment]
@jbphet - nevermind the previous request.

It looks like the file was checked out successfully before the xml file was built.

```
Feb 26 16:54:58 phet-server.int.colorado.edu build-server[22220]: info: pulling from babel
Feb 26 16:54:58 phet-server.int.colorado.edu build-server[22220]: debug: running git checkout master from ../babel
Feb 26 16:54:58 phet-server.int.colorado.edu build-server[22220]: debug: stderr: Already on 'master'
Feb 26 16:54:58 phet-server.int.colorado.edu build-server[22220]: info: git pull on babel
Feb 26 16:54:58 phet-server.int.colorado.edu build-server[22220]: debug: running git pull from ../babel
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: debug: stderr: From https://github.com/phetsims/babel
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: 825c6e4..33ca732  master     -> origin/master
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: debug: stdout: Updating 825c6e4..33ca732
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: Fast-forward
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: debug: stdout:  joist/joist-strings_sv.json                        |   7 +
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: .../plinko-probability-strings_sv.json             | 170 +++++++++++++++++++++
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: 2 files changed, 177 insertions(+)
Feb 26 16:54:59 phet-server.int.colorado.edu build-server[22220]: create mode 100644 plinko-probability/plinko-probability-strings_sv.json
```

**Comment by mattpen** [comment]
OK, so I believe that the swedish translation was skipped because the sim file did not exist yet in the deploy directory. This is likely due to the fact that the copy command that runs after grunt is using a callback rather than promises.  This command needs to execute synchronously.  I'll wrap it in an async/await, I think that should fix the problem.

**Comment by jbphet** [comment]
@mattpen - if you like, you can assign this to me to keep an eye on whether this is recurs in the next couple of weeks or so.  I tend to monitor the translation submissions on an almost-daily basis anyway.

**Comment by mattpen** [comment]
@jbphet this has not been deployed yet. I need to test it first, I'll notify you when it is up (probably Monday 3/5)

**Comment by jbphet** [comment]
I just checked the logs, and there are two more translations that have been submitted that are not appearing on the site.  Both are for the 'mr' locale (Marathi), and the sims are unit-rates and states-of-matter-basics.  @mattpen - can you please work your magic and get these to appear on the website?

**Comment by jbphet** [comment]
Two more appeared yesterday, both also with the 'mr' locale (Marathi), the sims are reactants-products-and-leftovers and rutherford-scattering.

**Comment by mattpen** [comment]
The fix was deployed as part of #87.  I'll manually update the few mr sims that fell through the cracks.

**Comment by mattpen** [comment]
I was able to synchronize the simulations listed above for `mr` with the website and confirm that they are published.  I was also able to add the translation credits and verify they were added to the database, but I could not verify that they were published on the website.  I believe they will show up in a day or so once the various caches time out.
